services/hanke-service/src/test/kotlin/fi/hel/haitaton/hanke/factory/AlluDataFactory.kt

```diff
@@ -101,20 +103,22 @@ class AlluDataFactory(val applicationRepository: ApplicationRepository) {
             workDescription: String = "Work description.",
             contractorWithContacts: CustomerWithContacts =
                 CustomerWithContacts(createCompanyCustomer(), listOf(createContact())),
+            rockExcavation: Boolean = false,
         ) =
             CableReportApplicationData(
-                ApplicationType.CABLE_REPORT,
-                name,
-                customerWithContacts,
-                geometry,
-                startTime,
-                endTime,
-                pendingOnClient,
-                identificationNumber,
-                clientApplicationKind,
-                workDescription,
-                contractorWithContacts,
-                false
+                applicationType = ApplicationType.CABLE_REPORT,
```

Comment:
https://sonarcloud.io/project/issues?resolved=false&severities=MAJOR&sinceLeakPeriod=true&types=CODE_SMELL&pullRequest=262&id=City-of-Helsinki_haitaton-backend&open=AYY2viW8HPUzc9biennP

Sonar complains about too many functions parameters. Would it be possible (or make sense) to wrap some of these parameters to a separate class?

Reply:
I don't think it makes much sense to wrap the parameters of a factory function to an object. The function exists to set defaults to the created object.